code/data_structures/tree_topview.cpp

```diff
@@ -0,0 +1,62 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+struct node
+{
+	int data;
+	struct node* left;
+	struct node* right;	
+};
+
+struct node* insertNode(int data)
+{
+	struct node* temp=(struct node*)malloc(sizeof(struct node));
+	temp->data=data;
+	temp->left=NULL;
+	temp->right=NULL;
+	return temp;
+}
+
+void topView(struct node* root)
+{
+	if(root==NULL)
+	return;
+	
+	map<int,int> m;
+	queue<pair<node*,int> > q;
+	q.push(make_pair(root,0));
+	while(!q.empty())
+	{
+		pair<node*,int> temp=q.front();
+		q.pop();
+		struct node* root=temp.first;
+		int val=temp.second;
+		if(m.find(val)==m.end())
+		m[val]=root->data;
+		if(root->left!=NULL)
+		q.push(make_pair(root->left,val-1));
+		if(root->right!=NULL)
+		q.push(make_pair(root->right,val+1));
+	}
+	map<int,int> :: iterator it;
+	for(it=m.begin();it!=m.end();it++)
+	{
+		printf("%d ",m[it->first]);
```

Comment:
Why use printf when you include `<iostream>`?
This program does not compile using Clang 4.0.0 because printf is in cstdio